pkg/util/utils.go

```diff
@@ -187,10 +187,10 @@ func CommandExecutesSuccessfully(cmd string, args ...string) bool {
 func IsDirectoryWritable(path string) bool {
 	tmpFilePath := filepath.Join(path, "tmp")
 	_, err := os.Create(tmpFilePath)
-	defer os.Remove(tmpFilePath)
 	if err != nil {
 		return false
 	}
+	defer os.Remove(tmpFilePath)
```

Comment:
Why move this line?

Reply:
@anjannath Because we should handle the error from the `os.Create` and then call anything else. It is just more logical.

Reply:
it does not make sense to move it... defer is just acting like a queue

Reply:
It is not having significant impact here but actually it does if we defer the function:
#### When defer is after error check
```
  _, err := os.Create("/usr/bin/tmp")
  if err != nil {
    fmt.Println("err before defer", err)
    return false
  }
  defer func(){
    fmt.Println("err in defer: ", err)
    os.Remove(tmpFilePath)
  }()

  # Output
  err before defer open /usr/bin/tmp: permission denied
```     

#### When defer is before error check
```
  _, err := os.Create("/usr/bin/tmp")
  defer func(){
    fmt.Println("err in defer: ", err)
    os.Remove(tmpFilePath)
    }()
  if err != nil {
    fmt.Println("err before defer", err)
    return false
  }

  # Output
  err before defer open /usr/bin/tmp: permission denied
  err in defer:  open /usr/bin/tmp: permission denied
```

Check https://play.golang.org/p/pluqU0OQlEq